### DISCUSSION TITLE
Changing histosys interpolation codes

### DISCUSSION THREAD

**Discussion by iokarkan** [scikit-hep/pyhf#1393] [Q&A] [answered]
I am trying to explore the effect of different interpolation codes in histosys modifiers, and I am believe these should be already implemented, however I cannot find anywhere in the documentation any information on how to set them.

I have looked at the source, and tried experimenting with the parameter configuration in the measurement, or as kwargs in the pyhf.infer.mle.fit method, but to no avail.

Any help is much appreciated!

**Comment by alexander-held** [comment, marked as answer]
You can set them when creating the model, example:
```python
workspace = pyhf.Workspace(spec)
model = workspace.model(
    modifier_settings={"normsys": {"interpcode": "code4"}, "histosys": {"interpcode": "code4p"}}
)
```
These are the default settings.

The available interpolations are `"code0", "code1", "code2", "code4", "code4p"`. I did not see this documented in the [`pyhf.Model` documentation](https://pyhf.readthedocs.io/en/v0.6.1/_generated/pyhf.pdf.Model.html) or under `_ModelConfig`, so it might be a good idea to add it there (or maybe under [Interpolators](https://pyhf.readthedocs.io/en/v0.6.1/api.html#interpolators)).

**Reply by kratsg** [reply]
#1425 1425
